packages/cf.js/src/provider.ts

```diff
@@ -40,43 +47,150 @@ export class Provider {
   ): Promise<Node.MethodResponse> {
     const requestId = cuid();
     return new Promise<Node.MethodResponse>((resolve, reject) => {
+      const request: Node.MethodRequest = {
+        requestId,
+        params,
+        type: methodName
+      };
       this.requestListeners[requestId] = response => {
         if (response.type === Node.ErrorType.ERROR) {
-          return reject(response.data);
+          return reject({
```

Comment:
Is there a `MethodReject` interface that the `reject` param can be typed to?

Reply:
No. There might be once I think more about the types of errors that methods can produce